stl/inc/yvals.h

```diff
@@ -7,6 +7,11 @@
 #ifndef _YVALS
 #define _YVALS
 #include <yvals_core.h>
+
+#ifdef _ENFORCE_ONLY_CORE_HEADERS

```

Comment:
I think this should be checked within `_STL_COMPILER_PREPROCESSOR` - we want to expose as little machinery as possible to non-compiler tools.